scylla/src/transport/connection_pool.rs

```diff
@@ -1108,20 +1108,22 @@ impl PoolRefiller {
             .await
             .map_err(|_| QueryError::TimeoutError)?;
 
-            // If there was at least one Ok and the rest were IoErrors we can return Ok
+            // If there was at least one Ok and the rest were broken connection errors we can return Ok
             // keyspace name is correct and will be used on broken connection on the next reconnect
 
-            // If there were only IoErrors then return IoError
-            // If there was an error different than IoError return this error - something is wrong
+            // If there were only broken connection errors then return broken connection error.
+            // If there was an error different than broken connection error return this error - something is wrong
 
             let mut was_ok: bool = false;
-            let mut io_error: Option<Arc<std::io::Error>> = None;
+            let mut broken_conn_error: Option<QueryError> = None;
 
             for result in use_keyspace_results {
                 match result {
                     Ok(()) => was_ok = true,
                     Err(err) => match err {
-                        QueryError::IoError(io_err) => io_error = Some(io_err),
+                        QueryError::IoError(_)
+                        | QueryError::BrokenConnection(_)
+                        | QueryError::ConnectionPoolError(_) => broken_conn_error = Some(err),
                         _ => return Err(err),
                     },
                 }
```

Comment:
This logic looks duplicated - see another place modified in this commit, in `cluster.rs`. Can we somehow deduplicate it?

Reply:
I think some iterator API could be used here to make it less verbose.

Reply:
Or maybe not. I'll just extract it to function